internal/operator-controller/controllers/clusterextension_controller.go

```diff
@@ -157,6 +161,17 @@ func ensureAllConditionsWithReason(ext *ocv1.ClusterExtension, reason v1alpha1.C
 	}
 }
 
+// isDeprecationCondition reports whether the given type is one of the deprecation
+// conditions we manage separately.
+func isDeprecationCondition(condType string) bool {
```

Comment:
nit: I would rather add this helper function to `api/v1/clusterextension_types.go`